wdl-ast/src/v1/task/common/container/value.rs

```diff
@@ -0,0 +1,407 @@
+//! Values for the `container` item within the `runtime` and `requirements`
```

Comment:
This gets, admittedly, a bit more gnarly than I would have hoped. I iterated on quite a few designs on how to capture all of the information I wanted. To summarize,

- I landed on calling the entire right side of the `container` assignment as a `Value`. The `Value` can be `Kind::String` (a single, string literal) or `Kind::Array` (an array of string literals).
- Each individual string literal within the value is represented by a `Uri`. The `Uri` can be `Kind::Any` (representing the `*` container URI) or `Kind::Entry` (open to a better name here, but represents any defined container URI that is not `*`).

Both `Value` and `Uri` are dynamically typed because (a) they both have common functionality where they keep track of the backing `AstNode` regardless of their type and (b) it was easier to test the `Kind`s without needing to deal with the fact that both `Value` and `Uri` need backing `AstNode`s.